core/spec.md

```diff
@@ -333,6 +366,13 @@ Therefore, the hierarchical structure of the Registry model is defined in such
 a way that it can be represented in a single file, including but not limited
 to JSON, or via the entity graph of a REST API.
 
+If an error occurs during the processing of a request, even if the error was
+during the creation of the response (e.g. an invalid `?inline` value was
+provided), then an error MUST be generated and the entire request (even any
+update actions) MUST be undone. In general, when an error is generated it
```

Comment:
Is there a transactionality element here, in terms of requiring that no changes applied due to the processing of the request must be observable until the processing has completed? (i.e. you can't just start making changes which could be observed, then revert them later).

Reply:
I kind of think there has to be, at least on a per API call basis, otherwise we run the risk of people seeing a Registry that's in a bad/invalid state.

Perhaps we could add a section to the spec (or primer) to talk about how:
- our expectation is that there's a transaction system in place where each individual API call is processed within the scope of one Tx such that if something goes wrong during ANY of its processing then entire thing can be rolled back
- this also then implies that any changes being made within that one Tx are not visible to other API calls (Txs) until it is committed 
- if someone wants to expose parts of their "desired set of changes" in an incremental fashion, they can do so by breaking up the work into individual API calls. 

Thoughts?